### PR TITLE
Send test results to BuildPulse for macOS job

### PR DESCRIPTION
**Pull request by jasonrudolph**
Example build: https://github.com/jasonrudolph/brew/runs/2862661764

Screen Shot 2021-06-18 at 8 52 54 PM

---

/cc @MikeMcQuaid for early 👀

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -229,6 +232,16 @@ jobs:
...
 
+      - name: Upload test results to BuildPulse for flaky test detection
+        if: '${{ matrix.buildpulse }} && !cancelled()' # Run this step even when the tests fail. Skip if the workflow is cancelled or BuildPulse isn't enabled for matrix job.
+        uses: Workshop64/buildpulse-action@main
+        with:
+          account: 2988
+          repository: 377629038
+          path: ./Library/Homebrew/test/junit
+          key: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
+          secret: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml]: I guess this is the tricky bit as it wouldn't work for PRs from forks. Thoughts?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -189,10 +189,13 @@ jobs:
         include:
           - name: tests (no-compatibility mode)
             test-flags: --no-compat --online --coverage
+            buildpulse: false
           - name: tests (generic OS)
             test-flags: --generic --online --coverage
+            buildpulse: true
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml]: Might want to pick the macOS job for this because it runs the most tests (e.g. `brew cask` ones)

Comment (jasonrudolph) [reply .github/workflows/tests.yml]: Cool. That's done. 

Note that it includes [this hack](https://github.com/jasonrudolph/brew/pull/2/files#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fR272-R275) just for the purposes of getting things to run within this fork. That hack won't be part of the upstream PR. 😇

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jasonrudolph) [reply .github/workflows/tests.yml]: That's right. Although now that the BuildPulse uploads have moved to the macOS job (which only runs on Homebrew/brew and doesn't run in forks), I think that resolves the issue. Does that sound right?

Note: For testing/demonstration purposes, this PR [enables the macOS jobs for all forks here](https://github.com/jasonrudolph/brew/pull/2/files#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fL249), but that change won't be part of the upstream PR.

**Comment by jasonrudolph** [comment]
Closing this out since it's now present in Homebrew/brew as of https://github.com/Homebrew/brew/commit/41803ebf935101ee3b87edc3db9e146d90d7ce15.

**Pull request closed without merging**